onadata/apps/logger/management/commands/soft_delete_orphan_attachments.py

```diff
@@ -43,11 +63,58 @@ def handle(self, *args, **kwargs):
                 self.stdout.write(
                     f'Processing Instance object #{instance_id}{message}…'
                 )
-            soft_deleted_attachments = get_soft_deleted_attachments(instance)
+
+            if start_date and instance.date_created < datetime_from_str(
+                f'{start_date}T00:00:00 +0000'
+            ):
```

Comment:
For future reference: I used the python way to make avoid `JOIN` as much as possible on those tables because the data is pretty big and it was slow to start the loop. 
But, ORM lets us skip 95% of non-edited submissions even it can take several hours before the queryset returns something (on production servers).